src/engine.rs

```diff
@@ -337,7 +337,13 @@ impl Reedline {
             (true, _) => {
                 self.history.set_navigation(HistoryNavigationQuery::Normal);
             }
-            (false, HistoryNavigationQuery::PrefixSearch(_)) => {}
+            (false, HistoryNavigationQuery::PrefixSearch(prefix)) => {
```

Comment:
Out of curiosity: Against which state is this guarding?

Reply:
Each up/down press will call this function. if the prefix does not change we don't want to reset the history state.